src/services/proposal.service.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Repository } from 'typeorm';
+
+import { ProposalEntity } from '@app/database';
+
+@Injectable()
+export class ProposalService {
+    constructor(@InjectRepository(ProposalEntity) private readonly _repository: Repository<ProposalEntity>) {}
+
+    getById = async (id: number): Promise<ProposalEntity> => {
+        return await this._repository.findOne({ where: { id } });
+    };
+
+    fetch = async (): Promise<ProposalEntity[]> => {
+        return this._repository.find({
+            order: {
+                id: 'DESC',
+            },
+        });
+    };
+
+    createOrUpdateProposal = async (data: Partial<ProposalEntity>) => {
```

Comment:
Could we use the `findOneAndUpdate` from TypeOrm ?
```
 createOrUpdateProposal = async (data: Partial<ProposalEntity>) => {
        const proposal = await this._repository.findOneAndUpdate({ id: data.id }, { $set: data }, { new: true, upsert: true });
        return proposal;
    };
```

Reply:
Update won't create, not the same behavior